Zoom/Zoom7zUnarchiver.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/python
+#
+# Copyright 2010 Per Olofsson
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""See docstring for Zoom7zUnarchiver class"""
+
+import os
+import subprocess
+import shutil
+
+from autopkglib import Processor, ProcessorError
+
+
+__all__ = ["Zoom7zUnarchiver"]
+
+EXTNS = {
+    'zip': ['zip'],
+    'tar_gzip': ['tar.gz', 'tgz'],
+    'tar_bzip2': ['tar.bz2', 'tbz'],
+    'tar': ['tar'],
+    'gzip': ['gzip'],
+    '7z': ['7z'],
+}
+
+class Zoom7zUnarchiver(Processor):
+    """Archive decompressor for zip and common tar-compressed formats."""
+    description = __doc__
+    input_variables = {
+        "archive_path": {
+            "required": False,
+            "description": "Path to an archive. Defaults to contents of the "
+                           "'pathname' variable, for example as is set by "
+                           "URLDownloader.",
+        },
+        "destination_path": {
+            "required": False,
+            "description": ("Directory where archive will be unpacked, created "
+                            "if necessary. Defaults to RECIPE_CACHE_DIR/NAME.")
+        },
+        "purge_destination": {
+            "required": False,
+            "description": "Whether the contents of the destination directory "
+                           "will be removed before unpacking.",
+        },
+        "archive_format": {
+            "required": False,
+            "description": ("The archive format. Currently supported: 'zip', "
+                            "'tar_gzip', 'tar_bzip2', 'tar'. If omitted, the "
+                            "file extension is used to guess the format.")
+        },
```

Comment:
Any reason you are leaving in the support for all these other formats since this is a custom processor to unarchive 7z -- and yet the description doesn't mention that?